packages/commonwealth/client/scripts/controllers/chain/ethereum/generalContracts.ts

```diff
@@ -0,0 +1,227 @@
+    to: string,
+    data: string,
+    wallet: IWebWallet<any>
+  ): Promise<TransactionReceipt> {
+    // Not using contractApi because it's ethers-dependent
+    // Non hardhat, non ethers Web3 Lib solution for signing and submitting tx
+    return this.sendTransaction(wallet.api, {
+      from: wallet.accounts[0],
+      to,
+      data,
+    });
+  }
+
+  public async callContractFunction(
+    fn: AbiItem,
+    processedArgs: any[],
+    wallet: IWebWallet<any>
+  ): Promise<TransactionReceipt | string> {
+    const methodSignature = `${fn.name}(${fn.inputs
+      .map((input) => input.type)
+      .join(',')})`;
+
+    const functionContract = this.web3Contract;
+    const contract = this.contract;
+
+    const functionTx = functionContract.methods[methodSignature](
+      ...processedArgs
+    );
+    if (fn.stateMutability !== 'view' && fn.constant !== true) {
+      // Sign Tx with PK if not view function
+      const txReceipt: TransactionReceipt = await this.makeContractTx(
+        contract.address,
+        functionTx.encodeABI(),
+        wallet
+      );
+      return txReceipt;
+    } else {
+      // send transaction
+      const tx: string = await this.makeContractCall(
+        contract.address,
+        functionTx.encodeABI(),
+        wallet
+      );
+      return tx;
+    }
+  }
+
+  public decodeTransactionData(fn: AbiItem, tx: any): any[] {
+    // simple return type
+    let result;
+    if (fn.outputs.length === 1) {
+      let decodedTx;
+      if (
+        this.contract.nickname === 'curated-factory-goerli' &&
+        fn.name === 'createProject'
+      ) {
+        const eventAbiItem = parseEventFromABI(
+          this.contract.abi,
+          'ProjectCreated'
+        );
+        const decodedLog = this.web3.eth.abi.decodeLog(
+          eventAbiItem.inputs,
+          tx.logs[0].data,
+          tx.logs[0].topics
+        );
+        console.log('decodedLog', decodedLog);
+        decodedTx = decodedLog.projectAddress;
+      } else {
+        decodedTx = this.web3.eth.abi.decodeParameter(
+          fn.outputs[0].type,
+          tx
+        );
+      }
+      result = [];
+      result.push(decodedTx);
+    } else if (fn.outputs.length > 1) {
+      const decodedTxMap = this.web3.eth.abi.decodeParameters(
+        fn.outputs.map((output) => output.type),
+        tx
+      );
+      // complex return type
+      result = Array.from(Object.values(decodedTxMap));
+    }
+    return result;
+  }
+
+  public async createFactoryDao(
+    fn: AbiItem,
+    processedArgs: any[],
+    wallet: IWebWallet<any>,
+    daoForm: CreateFactoryEthDaoForm
+  ) {
+    const functionContract = this.web3Contract;
+
+    const methodSignature = `${fn.name}(${fn.inputs
+      .map((input) => input.type)
+      .join(',')})`;
+
+    const functionTx = functionContract.methods[methodSignature](
+      ...processedArgs
+    );
+
+    const contract = this.contract;
+
+    if (contract.nickname === 'curated-factory-goerli') {
```

Comment:
This is required. We don't have a good way yet to figuring out how to parse Events data that we need after executing write txs on Web3. I hardcoded this because we know which event we have to watch for for Curated Factories aka the ProjectCreated Event. From there we can grab the project address then create a new community. We don't have a good way of generalizing this yet. Dao Launching will require further iteration to generalize this further. I've attached a notion task that we should undertake here: https://www.notion.so/hicommonwealth/Generalize-retrieval-of-needed-output-data-after-executing-write-Txs-on-Web3-d9354e1890fd411ca395ad0eea7ac336

Reply:
This + above hardcoded specialized logic should be refactored out into a separate function just for Projects then, rather than being inlined here -- so we easily know what the conditional is doing + can easily change it.

Reply:
This is not resolved -- these two functions have nothing to do with the above and should go elsewhere.